marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/reader/ReadAllDataTypesTest.java

```diff
@@ -0,0 +1,91 @@
+package com.marklogic.hub.spark.sql.sources.v2.reader;
+
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.InputStreamHandle;
+import com.marklogic.hub.HubClient;
+import com.marklogic.hub.spark.sql.sources.v2.Options;
+import com.marklogic.io.Base64;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.types.DataTypes;
+import org.junit.jupiter.api.Test;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ReadAllDataTypesTest extends AbstractSparkReadTest{
+
+    @Test
+    void readAllDataTypes() {
+        runAsDataHubDeveloper();
+        loadTDE("User");
+
+        HubClient client = runAsDataHubOperator();
+        InputStreamHandle handle = new InputStreamHandle(readInputStreamFromClasspath("entityInstances/User/user1.json"));
+        handle.withFormat(Format.JSON);
+        client.getFinalClient().newJSONDocumentManager().write("/doc1.json", new DocumentMetadataHandle()
+                .withCollections("User")
+                .withPermission("data-hub-operator", DocumentMetadataHandle.Capability.READ, DocumentMetadataHandle.Capability.UPDATE),
+            handle);
+
+
+        Options options = newOptions().withView("User").withNumPartitions("2");
+        HubDataSourceReader dataSourceReader = new HubDataSourceReader(options.toDataSourceOptions());
+        List<InternalRow> rows = readRows(dataSourceReader);
+        assertEquals(1, rows.size());
+        for (InternalRow row : rows) {
```

Comment:
For clarity, just do "InternalRow row : rows.get(0)" since we know there's only one.